nvflare/app_common/executors/task_script_runner.py

```diff
@@ -71,18 +88,22 @@ def get_script_full_path(self, script_path) -> str:
                 if absolute_path.endswith(script_path):
                     parent_dir = absolute_path[: absolute_path.find(script_path)].rstrip(os.sep)
                     if os.path.isdir(parent_dir):
-                        target_file = absolute_path
-                        break
+                        path_components = parent_dir.split(os.path.sep)
+                        if site_name in path_components:
+                            target_file = absolute_path
+                            break
```

Comment:
what happens if user pass in "site_name"="site-1" and script_path="/users/YT/script.py"?